### ISSUE TITLE
Error in RDF Export with non-latin instance names

### ISSUE THREAD

**Issue by antohabio** [SemanticMediaWiki/SemanticMediaWiki#2188]
### Setup and configuration

- SMW version: 2.4.5
- MW version: 1.28
- PHP version: 5.6.24-1
- DB (MySQL etc.): MySQL

### Issue

I have error after attempt to export RDF from page (example, page):

> This page contains the following errors:
> error on line 38 at column 9: Failed to parse QName 'wiki:'
> Below is a rendering of the page up to the first error.

Source code of this error-page (example, page):

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE rdf:RDF[
	<!ENTITY rdf 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'>
	<!ENTITY rdfs 'http://www.w3.org/2000/01/rdf-schema#'>
	<!ENTITY owl 'http://www.w3.org/2002/07/owl#'>
	<!ENTITY swivt 'http://semantic-mediawiki.org/swivt/1.0#'>
	<!ENTITY wiki 'http://anthecology.org/index.php?title=&#37;D0&#37;A1&#37;D0&#37;BB&#37;D1&#37;83&#37;D0&#37;B6&#37;D0&#37;B5&#37;D0&#37;B1&#37;D0&#37;BD&#37;D0&#37;B0&#37;D1&#37;8F:URIResolver/'>
	<!ENTITY category 'http://anthecology.org/index.php?title=&#37;D0&#37;A1&#37;D0&#37;BB&#37;D1&#37;83&#37;D0&#37;B6&#37;D0&#37;B5&#37;D0&#37;B1&#37;D0&#37;BD&#37;D0&#37;B0&#37;D1&#37;8F:URIResolver/Category-3A'>
	<!ENTITY property 'http://anthecology.org/index.php?title=&#37;D0&#37;A1&#37;D0&#37;BB&#37;D1&#37;83&#37;D0&#37;B6&#37;D0&#37;B5&#37;D0&#37;B1&#37;D0&#37;BD&#37;D0&#37;B0&#37;D1&#37;8F:URIResolver/Property-3A'>
	<!ENTITY wikiurl 'http://anthecology.org/index.php?title='>
]>

<rdf:RDF
	xmlns:rdf="&rdf;"
	xmlns:rdfs="&rdfs;"
	xmlns:owl ="&owl;"
	xmlns:swivt="&swivt;"
	xmlns:wiki="&wiki;"
	xmlns:category="&category;"
	xmlns:property="&property;">

	<owl:Ontology rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:ExportRDF/0000007656">
		<swivt:creationDate rdf:datatype="http://www.w3.org/2001/XMLSchema#dateTime">2017-01-18T10:40:58+03:00</swivt:creationDate>
		<owl:imports rdf:resource="http://semantic-mediawiki.org/swivt/1.0"/>
	</owl:Ontology>
	<swivt:Subject rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/0000007656">
		<rdf:type rdf:resource="&wiki;-D0-9A-D0-B0-D1-82-D0-B5-D0-B3-D0-BE-D1-80-D0-B8-D1-8F-3A-D0-9F-D1-83-D0-B1-D0-BB-D0-B8-D0-BA-D0-B0-D1-86-D0-B8-D0-B8"/>
		<rdfs:label>0000007656</rdfs:label>
		<swivt:page rdf:resource="http://anthecology.org/index.php?title=0000007656"/>
		<rdfs:isDefinedBy rdf:resource="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:ExportRDF/0000007656"/>
		<swivt:wikiNamespace rdf:datatype="http://www.w3.org/2001/XMLSchema#integer">0</swivt:wikiNamespace>
		<property:AID rdf:resource="http://anthecology.org/index.php?title=0000007656"/>
		<property:DOI rdf:resource="https://doi.org/10.1657/1523-0430-41.1.103"/>
		<property:Has_query rdf:resource="&wiki;0000007656-23_QUERY888cc67cabba666c6c34302c316034d3"/>
		<swivt:wikiPageModificationDate rdf:datatype="http://www.w3.org/2001/XMLSchema#dateTime">2017-01-17T18:51:36Z</swivt:wikiPageModificationDate>
		<property:Modification_date-23aux rdf:datatype="http://www.w3.org/2001/XMLSchema#double">2457771.2858333</property:Modification_date-23aux>
		<swivt:wikiPageSortKey rdf:datatype="http://www.w3.org/2001/XMLSchema#string">0000007656</swivt:wikiPageSortKey>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-90-D0-B2-D1-82-D0-BE-D1-80 rdf:resource="&wiki;Garcia-2Dcamacho-2Dr"/>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-90-D0-B2-D1-82-D0-BE-D1-80 rdf:resource="&wiki;Totland-2Do"/>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-93-D0-BE-D0-B4 rdf:datatype="http://www.w3.org/2001/XMLSchema#gYear">2009</wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-93-D0-BE-D0-B4>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-93-D0-BE-D0-B4-23aux rdf:datatype="http://www.w3.org/2001/XMLSchema#double">2454832.5</wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-93-D0-BE-D0-B4-23aux>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-96-D1-83-D1-80-D0-BD-D0-B0-D0-BB rdf:resource="&wiki;Arctic-2Dantarctic-2Dand-2Dalpine-2Dresearch"/>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-97-D0-B0-D0-B3-D0-BE-D0-BB-D0-BE-D0-B2-D0-BE-D0-BA rdf:datatype="http://www.w3.org/2001/XMLSchema#string">Pollen Limitation in the Alpine: a Meta-Analysis</wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-97-D0-B0-D0-B3-D0-BE-D0-BB-D0-BE-D0-B2-D0-BE-D0-BA>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-A2-D0-B8-D0-BF_-D0-BF-D1-83-D0-B1-D0-BB-D0-B8-D0-BA-D0-B0-D1-86-D0-B8-D0-B8 rdf:resource="&wiki;-D0-A1-D1-82-D0-B0-D1-82-D1-8C-D1-8F"/>
		<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-A7-D0-B8-D1-81-D0-BB-D0-BE_-D1-81-D1-82-D1-80-D0-B0-D0-BD-D0-B8-D1-86 rdf:datatype="http://www.w3.org/2001/XMLSchema#double">1</wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-A7-D0-B8-D1-81-D0-BB-D0-BE_-D1-81-D1-82-D1-80-D0-B0-D0-BD-D0-B8-D1-86>
	</swivt:Subject>
	<swivt:Subject rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/0000007656-23_QUERY888cc67cabba666c6c34302c316034d3">
		<swivt:masterPage rdf:resource="&wiki;0000007656"/>
		<swivt:wikiNamespace rdf:datatype="http://www.w3.org/2001/XMLSchema#integer">0</swivt:wikiNamespace>
		<property:Query_depth rdf:datatype="http://www.w3.org/2001/XMLSchema#double">1</property:Query_depth>
		<property:Query_format rdf:datatype="http://www.w3.org/2001/XMLSchema#string">list</property:Query_format>
		<property:Query_size rdf:datatype="http://www.w3.org/2001/XMLSchema#double">3</property:Query_size>
		<property:Query_string rdf:datatype="http://www.w3.org/2001/XMLSchema#string">[[Категория:Публикации]] [[Исходящая ссылка::0000007656]]</property:Query_string>
		<swivt:wikiPageSortKey rdf:datatype="http://www.w3.org/2001/XMLSchema#string">0000007656</swivt:wikiPageSortKey>
	</swivt:Subject>
	<owl:DatatypeProperty rdf:about="http://semantic-mediawiki.org/swivt/1.0#creationDate" />
	<owl:Class rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-9A-D0-B0-D1-82-D0-B5-D0-B3-D0-BE-D1-80-D0-B8-D1-8F-3A-D0-9F-D1-83-D0-B1-D0-BB-D0-B8-D0-BA-D0-B0-D1-86-D0-B8-D0-B8" />
	<owl:ObjectProperty rdf:about="http://semantic-mediawiki.org/swivt/1.0#page" />
	<owl:DatatypeProperty rdf:about="http://semantic-mediawiki.org/swivt/1.0#wikiNamespace" />
	<owl:ObjectProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AAID" />
	<owl:ObjectProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3ADOI" />
	<owl:ObjectProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AHas_query" />
	<owl:DatatypeProperty rdf:about="http://semantic-mediawiki.org/swivt/1.0#wikiPageModificationDate" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AModification_date-23aux" />
	<owl:DatatypeProperty rdf:about="http://semantic-mediawiki.org/swivt/1.0#wikiPageSortKey" />
	<owl:ObjectProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-90-D0-B2-D1-82-D0-BE-D1-80" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-93-D0-BE-D0-B4" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-93-D0-BE-D0-B4-23aux" />
	<owl:ObjectProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-96-D1-83-D1-80-D0-BD-D0-B0-D0-BB" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-97-D0-B0-D0-B3-D0-BE-D0-BB-D0-BE-D0-B2-D0-BE-D0-BA" />
	<owl:ObjectProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-A2-D0-B8-D0-BF_-D0-BF-D1-83-D0-B1-D0-BB-D0-B8-D0-BA-D0-B0-D1-86-D0-B8-D0-B8" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-A7-D0-B8-D1-81-D0-BB-D0-BE_-D1-81-D1-82-D1-80-D0-B0-D0-BD-D0-B8-D1-86" />
	<owl:ObjectProperty rdf:about="http://semantic-mediawiki.org/swivt/1.0#masterPage" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AQuery_depth" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AQuery_format" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AQuery_size" />
	<owl:DatatypeProperty rdf:about="http://anthecology.org/index.php?title=%D0%A1%D0%BB%D1%83%D0%B6%D0%B5%D0%B1%D0%BD%D0%B0%D1%8F:URIResolver/Property-3AQuery_string" />
	<!-- Created by Semantic MediaWiki, https://www.semantic-mediawiki.org/ -->
</rdf:RDF>
```

The problem in **cyrillyc** name of Property, begin from:

`<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-90-D0-B2-D1-82-D0-BE-D1-80 rdf:resource="&wiki;Garcia-2Dcamacho-2Dr"/>`

**Comment by kghbln** [comment]
Thanks for reporting the issue on [your wiki](http://anthecology.org/index.php?title=%D0%97%D0%B0%D0%B3%D0%BB%D0%B0%D0%B2%D0%BD%D0%B0%D1%8F_%D1%81%D1%82%D1%80%D0%B0%D0%BD%D0%B8%D1%86%D0%B0&uselang=en)

@mwjames FYI

**Comment by mwjames** [comment]
> The problem in cyrillyc name of Property, begin from:
<wiki:-D0-A1-D0-B2-D0-BE-D0-B9-D1-81-D1-82-D0-B2-D0-BE-3A-D0-90-D0-B2-D1-82-D0-BE-D1-80 rdf:resource="&wiki;Garcia-2Dcamacho-2Dr"/>

This problem arises because of `$wgLanguageCode = "ru";`, if you were to use `$wgLanguageCode = "en";` the encoding would not be distorted which doesn't mean this is correct or the anticipated behaviour.

The issue emerges because the `Escaper` tries to decode the `Property` NS and ends up with a -D... encoded string (in the `ru` content language) which of course creates an invalid `XML` (as it starts with a `-`). Now, this particular code has been around at least since 1.8.x [1] therefore this is not a regression of the 2.4 release or its predecessors.

Further more, other non-latin languages (as content language) will face a similar issue hence this is not just a simple "fix the encode strategy" but a matter of IRI support in general therefore it cannot be fixed in the 2.4 branch and it will most likely require a flag to switch to a different encode mode to avoid BC breakages for the existing RDF export format.

Some material on the topic of "non-ASCII characters" support for property instance names and resource identifiers are available at [2, 3, 4, 5].

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/src/Exporter/Escaper.php#L31
[1]https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/1.8.0.5/includes/export/SMW_Exporter.php#L289-L304
[2] http://svn.aksw.org/papers/2010/ISWC_I18n/public.pdf
[3] https://www.w3.org/TR/rdf-syntax-grammar
[4] https://www.w3.org/TR/rdf11-concepts/#section-IRIs
[5] http://dcevents.dublincore.org/IntConf/dc-2011/paper/download/47/15

**Comment by antohabio** [comment]
Thanks for the answer.

Mechanism error I knew about.
I tried to change $wgLanguageCode = "ru"; to $wgLanguageCode = "en";, error has really disappeared.
In principle, I would change it to English, but then no longer work previously created, such as category (Категория:Название does not work; it becomes work Category:Название).
My English is not very good, but if I understand correctly, then:
- currently not fix the problem;
- switching to English - a crutch.
Is it possible to fix this in future versions?

**Comment by mwjames** [comment]
> I would change it to English, but then no longer work previously created, such as category 

... and this is why that isn't really a solution.

> currently not fix the problem

Not for the 2.4 branch.

> Is it possible to fix this in future versions?

This is what I meant with "...it will most likely require a flag to switch to a different encode mode to avoid BC breakages for the existing RDF export format ...", I'll try to get something for the 2.5 branch using an experimental flag to avoid breaking the existing RDF export.

**Comment by antohabio** [comment]
Sincerely grateful for your help.

**Comment by mwjames** [comment]
>  I'll try to get something for the 2.5 branch using an experimental flag to avoid breaking the existing RDF export.

Addressed in #2207.
